descartes_trajectory/src/axial_symmetric_pt.cpp

```diff
@@ -0,0 +1,46 @@
+#include "descartes_trajectory/axial_symmetric_pt.h"
+
+
+using descartes_trajectory::TolerancedFrame;
+using descartes_trajectory::AxialSymmetricPt;
+using namespace descartes_core::utils;
+
+static TolerancedFrame makeRotationalAxis(AxialSymmetricPt::FreeAxis axis)
+{
+  using namespace descartes_trajectory;
+
+  Eigen::Affine3d rot = Eigen::Affine3d::Identity();
+  PositionTolerance pos_tol = ToleranceBase::zeroTolerance<PositionTolerance>(0,0,0);
+  OrientationTolerance orient_tol = ToleranceBase::createSymmetric<OrientationTolerance>(0.0, 0.0, 0.0, 
+                                    ((axis == AxialSymmetricPt::X_AXIS) ? 2*M_PI : 0.0),
+                                    ((axis == AxialSymmetricPt::Y_AXIS) ? 2*M_PI : 0.0),
+                                    ((axis == AxialSymmetricPt::Z_AXIS) ? 2*M_PI : 0.0));
+  return TolerancedFrame(rot, pos_tol, orient_tol);
+} 
+
+
+namespace descartes_trajectory
+{
+
+AxialSymmetricPt::AxialSymmetricPt(double x, double y, double z, double rx, double ry, double rz,
+                                   double orient_increment, FreeAxis axis) :
+  CartTrajectoryPt(toFrame(x, y, z, rx, ry, rz, EulerConventions::XYZ),
+                   makeRotationalAxis(axis),
+                   Eigen::Affine3d::Identity(),
+                   Eigen::Affine3d::Identity(),
+                   0.0, // The position discretization
+                   orient_increment) // Orientation discretization (starting at -2Pi and marching to 2Pi)
+{
+}
+
+AxialSymmetricPt::AxialSymmetricPt(const Eigen::Affine3d& pose, double orient_increment, FreeAxis axis) :
+  CartTrajectoryPt(pose,
+                   makeRotationalAxis(axis),
+                   Eigen::Affine3d::Identity(),
+                   Eigen::Affine3d::Identity(),
+                   0.0, // The position discretization
+                   orient_increment) // Orientation discretization (starting at -2Pi and marching to 2Pi)
+{
+}
+
+} // end of namespace descartes_trajectory
```

Comment:
In the original version of the AxialSymetricPt the wobj_pt_ was initialized to the x,y,z,rx,ry,0 and wobj_base_ was set to Identity.  This resulted in fixed values for the translation and the rotations about x and y; only the rotation about z was discretized in the uniform method of the CartTrajectoryPt.  But, yes since the wobj_base now holds the nominal pose then the result should be equivalent, at least when rotating freely about z.  I think this new implementation is more flexible than the original one.